### DISCUSSION TITLE
Wrong balance after rescan with unconfirmed transactions

### DISCUSSION THREAD

**Discussion by gus4rs** [btcpayserver/btcpayserver#6184] [Technical Support]
My wallet had 1 UTXO and I did a BTC transaction that spends this UTXO and sends 4 UTXOs to myself

While the transaction is sitting in the mempool, I imported that wallet in BTCPayServer using the 12 word seed.

After a wallet rescan, BTCPayserver is showing my wallet with 1 UTXO, clearly wrong, and ignoring the tx above

WTF is wrong with it?

**Comment by dennisreimann** [comment]
As long as the transaction is unconfirmed, the UTXO belongs to the spending wallet. What is wrong with that behaviour? Does it work as expected after the TX has been confirmed?

**Comment by gus4rs** [comment]
After it was confirmed, the tracker is working fine

Regarding the behaviour, it is incorrect: I started tracking a wallet that had 1 UTXO confirmed and 4 UTXOs unconfirmed. The tracker was returning empty for the unconfirmed transaction.

**Reply by pavlenex** [reply]
Did you rescan the wallet when you imported it?

**Reply by gus4rs** [reply]
Yes, more than once

**Comment by gus4rs** [comment]
To be fair, this is not a BTCPayserver issue, but a NBXplorer problem. It doesn't detect existent mempool transactions when you start tracking a derivation scheme
